src/naive/internals.rs

```diff
@@ -318,12 +319,6 @@ impl Of {
         self.0 >> 4
     }
 
-    #[inline]
-    pub(super) const fn with_ordinal(&self, ordinal: u32) -> Option<Of> {
-        let of = Of((ordinal << 4) | (self.0 & 0b1111));
-        of.validate()
```

Comment:
Previously there was no validation of an ordinal beyond this check. Because it happened after the shift left, some invalid values where silently accepted. I added a test.